### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by mrazauskas** [wellwelwel/poku#584] [Ideas]
Few ideas regarding benchmarks (very good to see them, of course!).

1. What if the same assertion library would be used with all runners? Any of them, but the same. Might be `node:assert`, or `chai`, or anything else. For instance, Mocha can be also used with `expect`, but it is slower than `node:assert`. I have a feeling that benchmark here should test speed of a test runner, not speed of assertion library.
2. Consider using [`hyperfine`](https://github.com/sharkdp/hyperfine) instead of spawning a process.

**Comment by wellwelwel** [comment]
I'll take a look at `hyperfine`. I like the idea of testing the test runner independently, but I was curious why not include the assertions as well.

For more transparency, the current tests were inspired by the `uvu` benchmark.

**Reply by wellwelwel** [reply]
Although **Poku** uses `assert` and `strict` from **Node.js**, it performs a parser that varies according to the platform, both before and after the assertion result, including a minimal builder:

https://github.com/wellwelwel/poku/blob/5e69de52c6b03d00d7255f7ad6becc06ce2c9af5/src/modules/essentials/assert.ts#L6

- That's why I believe it's also interesting to use `assert` and ensure that **Poku** set is well balanced.
